### DISCUSSION TITLE
How do I fix self-intersecting meshes?

### DISCUSSION THREAD

**Discussion by umar-hossain** [BrunoLevy/geogram#101] [Q&A] [answered]
Hi,

I recently downloaded and built Geogram using the instructions after seeing [this tweet](https://twitter.com/BrunoLevy01/status/1640427974007635998). How do I do the same thing with a tricky mesh of my own?

I can't seem to find any info in the wiki. Thank you for any help.

**Comment by BrunoLevy** [comment, marked as answer]
Hi, 
This feature is currently under development, normally it is reasonably robust, but you may still encounter some issues with it.
There is a new [example](https://github.com/BrunoLevy/geogram/tree/main/src/examples/geogram/intersect) that you can use like that:
```
$intersect inputmesh remove_internal_shells=true
```
The result is saved in `out.obj`
Do not hesitate to tell me if it does not work.
-- Bruno

**Reply by umar-hossain** [reply]
I'm not great at programming or using the command line - I have compiled GraphiteThree and loaded my faulty mesh into it.

I can see there is a command line at the bottom, but I can't seem to run the command there. If I type in "intersect remove_internal_shells=true" I get the following error:

![image](https://github.com/BrunoLevy/geogram/assets/77932256/a87d1aaa-ea67-484b-9ac9-dec72f1b9a90)

Is it possible to run this intersect command through GraphiteThree? Is it related to any of the tools in the repair mesh toolbar?
<img width="438" alt="image" src="https://github.com/BrunoLevy/geogram/assets/77932256/cb479b3b-e131-46ba-98ff-7d874a5df0e4">

**Comment by BrunoLevy** [comment]
- the command 'intersect' is in the "surfaces->boolean operations" menu (last one in the menu)
- when you do something from the GUI, use the up arrow on the keyboard in the terminal window to see how to do the same thing programatically
- you can create a '.lua' file with a program executed by Graphite
- more on Graphite scripting [here](https://github.com/BrunoLevy/GraphiteThree/wiki/scripting)

Hope this helps,
Best,
-- Bruno
